MekHQ/src/mekhq/gui/dialog/AddOrEditKillEntryDialog.java

```diff
@@ -47,29 +48,44 @@ public class AddOrEditKillEntryDialog extends JDialog {
     private int operationType;
     private Kill kill;
     private LocalDate date;
+    private int missionId;
+    private int scenarioId;
+    private Campaign campaign;
 
     private JButton btnClose;
     private JButton btnOK;
     private JLabel lblKill;
     private JTextField txtKill;
     private JLabel lblKiller;
     private JTextField txtKiller;
+    private JLabel lblDate;
     private JButton btnDate;
+    private JLabel lblMissionId;
+    private MMComboBox<Mission> cboMissionId;
+    private ArrayList<Integer> missionIdList;
+    private JLabel lblScenarioId;
+    private MMComboBox<String> cboScenarioId;
+    private ArrayList<Integer> scenarioIdList;
 
-    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, UUID killerPerson, String killerUnit, LocalDate entryDate) {
-        this(parent, modal, ADD_OPERATION, new Kill(killerPerson, "?", killerUnit, entryDate));
+    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, UUID killerPerson, String killerUnit, LocalDate entryDate, Campaign campaign) {
+        // We default missionId & scenarioId to 0 when adding new kills
+        this(parent, modal, ADD_OPERATION, new Kill(killerPerson, "?", killerUnit, entryDate, 0, 0), campaign);
     }
 
-    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, Kill kill) {
-        this(parent, modal, EDIT_OPERATION, kill);
+    public AddOrEditKillEntryDialog(JFrame parent, boolean modal, Kill kill, Campaign campaign) {
+        this(parent, modal, EDIT_OPERATION, kill, campaign);
     }
 
-    private AddOrEditKillEntryDialog(JFrame parent, boolean modal, int operationType, Kill kill) {
+    private AddOrEditKillEntryDialog(JFrame parent, boolean modal, int operationType, Kill kill, Campaign c) {
         super(parent, modal);
 
+        campaign = c;
+
         this.frame = parent;
         this.kill = Objects.requireNonNull(kill);
         this.date = this.kill.getDate();
+        this.missionId = this.kill.getMissionId();
```

Comment:
It doesn't look like there is good handling for null values here, which looks like a possibility if the default constructor were used.

Reply:
In the event the default constructor is used, `missionID` and `scenarioID` will == 0